### PR TITLE
Create forgot/reset password feature

### PR DESCRIPTION
**Pull request by jlvillegas04**
Chic@s, despues de mucho esfuerzo les presento la funcionalidad de Reset y Forgot password:
En el back tambien hice cambios, por favor verlos en el repo de back!
ms57-1
ms57-2

### CODE REVIEW

File: src/pages/Forgot/Forgot.jsx [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+      Swal.fire({
+        title: 'Email Sent',
+        text: 'An email has been sent to reset your password.',
+        icon: 'success',
+        confirmButtonText: 'Got It'
+      })
+    } catch (error) {
+      setMessage('An error occurred.')
+
+      // Show SweetAlert on error
```
Comment (ht1204) [inline src/pages/Forgot/Forgot.jsx:41]: this comment is not necessary.

File: src/pages/Reset/Reset.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+    // Check if passwords match
+    if (newPassword !== confirmPassword) {
+      setMessage('Passwords do not match.')
+      return
+    }
+    console.log('URL:', `/api/auth/reset-password?token=${token}`)
+    console.log('Token from URL:', token)
```
Comment (ht1204) [inline src/pages/Reset/Reset.jsx:26]: remove these console.logs, thx.

File: src/pages/Reset/Reset.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      <div className="reset-password__content">
+        <h1 className="reset-password__title">Reset Password</h1>
+        <form onSubmit={handleSubmit}>
+          <div className="reset-password__form-group">
+            <label className="reset-password__label">New Password:</label>
+            <input
+              className="reset-password__input"
+              type="password"
+              value={newPassword}
+              onChange={(e) => setNewPassword(e.target.value)}
```
Comment (ht1204) [inline src/pages/Reset/Reset.jsx:78]: also here, a handle callback event would be better.

File: src/pages/Forgot/Forgot.jsx [hunk 4]
```diff
@@ -1,8 +1,48 @@
...
+    console.log('handleSubmit function called')
+
+    try {
+      const response = await http.post('/auth/forgot-password', { email })
+      setMessage(response.data.message)
+
+      const templateParams = {
+        to_name: response.fullName,
+        to_email: email,
+        reset_link: `https://localhost:5173/reset-password?token=${response.data.token}`
```
[the remark this answers is not in the document]
Comment (erickfabiandev) [reply src/pages/Forgot/Forgot.jsx:23]: image
Juan imports the variable found in the host file and concatenates it to form the route

**Review by erickfabiandev — APPROVED** [review]

**Pull request merged**